src/db/mod.rs

```diff
@@ -374,17 +370,13 @@ impl ThreadedDatabase for Database {
     }
 
     fn drop_collection(&self, name: &str) -> Result<()> {
-        let mut spec = bson::Document::new();
-        spec.insert("drop", Bson::String(String::from(name)));
-        try!(self.command(spec, CommandType::DropCollection, None));
-        Ok(())
+        let spec = doc!{ "drop": name };
+        self.command(spec, CommandType::DropCollection, None).map(drop)
     }
 
     fn drop_database(&self) -> Result<()> {
-        let mut spec = bson::Document::new();
-        spec.insert("dropDatabase", Bson::I32(1));
-        try!(self.command(spec, CommandType::DropDatabase, None));
-        Ok(())
+        let spec = doc!{ "dropDatabase": 1_i32 };
```

Comment:
In this case, I don't think then type really matters; this can just be `1`.

Reply:
Right, let's do that.